plugins/modules/win_domain_ou.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: GPL-3.0-only
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r'''
+---
+module: win_domain_ou
+short_description: Manage Active Directory Organizational Units
+author: Joe Zollo (@joezollo), Larry Lane (@gamethis)
+requirements:
+  - This module requires Windows Server 2012 or Newer
+description:
+  - Manage Active Directory Organizational Units
+  - Adds, Removes and Modifies Active Directory Organizational Units
+  - Task should be delegated to a Windows Active Directory Domain Controller
+options:
+  name:
+    description:
+      - The name of the Organizational Unit
+    type: str
+    required: true
+  protected:
+    description:
+      - Indicates whether to prevent the object from being deleted. When this
+        l(protected=true), you cannot delete the corresponding object without
+        changing the value of the property.
+    type: bool
+    default: false
+  path:
+    description:
+      - Specifies the X.500 path of the OU or container where the new object is
+        created.
+      - defaults to adding ou at base of domain connected to.
+    type: str
+    required: false
+  state:
+    description:
+      - Specifies the desired state of the OU.
+      - When l(state=present) the module will attempt to create the specified
+        OU if it does not already exist.
+      - When l(state=absent), the module will remove the specified OU.
+      - When l(state=absent) and l(recursive=true), the module will remove all
+        the OU and all child OU's.
+    type: str
+    default: present
+    choices: [ present, absent ]
+  recursive:
+    description:
+      - Removes the OU and any child items it contains.
+      - You must specify this parameter to remove an OU that is not empty.
+    type: bool
+    default: false
+  domain_server:
+    description:
+    - Specifies the Active Directory Domain Services instance to connect to.
+    - Can be in the form of an FQDN or NetBIOS name.
+    - If not specified then the value is based on the domain of the computer
+      running PowerShell.
+    type: str
+  domain_username:
+    description:
+    - The username to use when interacting with AD.
+    - If this is not set then the user Ansible used to log in with will be
+      used instead when using CredSSP or Kerberos with credential delegation.
+    type: str
+  domain_password:
+    type: str
+    description:
+    - The password for the domain you are accessing
+  filter:
+    type: str
+    description: filter for lookup of ou.
+    default: '*'
+  properties:
+    type: dict
+    description:
+      - Defines specific LDAP properties for the organizational unit.
```

Comment:
You should describe this a bit more. Currently they are LDAP properties but it looks like the code is trying to convert the values to and from snake and camel case. Personally I think if this is going to be a free form field that it just use the camel case names that are in use with LDAP. It makes it easier for people wanting to set the raw values outside of the root module options without having to translate what they are actually for.